### ISSUE TITLE
[Question] Distribute apk through torrent

### ISSUE THREAD

**Issue by escape0707** [oxen-io/session-android#970]
Sorry for posting a question here, I don't know other place which suit this better.

Can we distribute all of those session client installer through a more government firewall blocking resistant way? Like through torrenting? When I need to send the latest installer to my friend or relative, I used to fire up `miniserve` but it's kinda cumbersome. If there is official torrent or even better RSS to follow, it will be easier.

**Comment by x80486** [comment]
This should be really straightforward to implement these days, and fits perfectly into the context that `Session` is in.

**Comment by hjubb** [comment]
It might make sense to have it available via a lokinet website in some way? Maybe we will integrate CI to build and distribute it the same way as other oxen related releases in the future. Could be a nice idea anyway to increase the availability

**Comment by escape0707** [comment]
> It might make sense to have it available via a lokinet website in some way? Maybe we will integrate CI to build and distribute it the same way as other oxen related releases in the future. Could be a nice idea anyway to increase the availability

Then how do they access lokinet in the first place without reaching to someone for the lokinet client?

**Comment by majestrate** [comment]
> It might make sense to have it available via a lokinet website in some way? Maybe we will integrate CI to build and distribute it the same way as other oxen related releases in the future. Could be a nice idea anyway to increase the availability

we could do it via bittorrent over lokinet too

**Comment by majestrate** [comment]
this issue is actually quite easy to do using [BEP 0019](https://www.bittorrent.org/beps/bep_0019.html), each release we can make a torrent that adds a webseed via github (if they allow it)

**Comment by hjubb** [comment]
> > It might make sense to have it available via a lokinet website in some way? Maybe we will integrate CI to build and distribute it the same way as other oxen related releases in the future. Could be a nice idea anyway to increase the availability
> 
> we could do it via bittorrent over lokinet too

Yeah it would tie alternate session discoverability to lokinet via a pretty hard dependency which would probably be an issue for most mobile clients

**Comment by escape0707** [comment]
Currently I'm telling my relatives to use F-Droid with a regional mirror hosted by a university as a workaround.

**Comment by hjubb** [comment]
I'm not sure if that would include the Session F-Droid repo which is external to the main repo, is there any difference with using the repo described here https://github.com/oxen-io/session-fdroid ? Or are you referring to github being blocked as a specific issue as well

**Comment by escape0707** [comment]
> I'm not sure if that would include the Session F-Droid repo which is external to the main repo, is there any difference with using the repo described here https://github.com/oxen-io/session-fdroid ? Or are you referring to github being blocked as a specific issue as well

Yeah, Google and GitHub are often blocked together.

https://f-droid.org/en/packages/network.loki.messenger.fdroid/

This fdroid version doesn't have FCM for fast notification, but since Google is blocked it doesn't matter anyway.
